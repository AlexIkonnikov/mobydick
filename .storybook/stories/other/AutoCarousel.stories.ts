import {Meta, StoryObj} from '@storybook/react-native';
import {Dimensions} from 'react-native';

import AutoCarouselExample from './Example/AutoCarouselExample';

import {ICarouselAlign} from '@lad-tech/mobydick-core';

const {width: WIDTH, height: HEIGHT} = Dimensions.get('screen');

const meta: Meta<typeof AutoCarouselExample> = {
  title: 'Other',
  component: AutoCarouselExample,
};

type Story = StoryObj<typeof AutoCarouselExample>;

export const AutoCarousel = {
  args: {
    sideMargin: 5,
    activeItemId: 1,
    dataLength: 3,
    align: ICarouselAlign.start,
    timerAuto: 2000,
    itemWidth: WIDTH - 10,
    itemHeight: HEIGHT,
    animateAutoScroll: true,
    isDots: false,
  },
  argTypes: {
    align: {
      options: Object.values(ICarouselAlign),
      control: {
        type: 'select',
      },
    },
    animateAutoScroll: {
      control: {
        type: 'boolean',
      },
    },
    isDots: {
      control: {
        type: 'boolean',
      },
    },
  },
} satisfies Story;

export default meta;
