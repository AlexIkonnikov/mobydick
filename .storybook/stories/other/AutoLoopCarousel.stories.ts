import {Meta, StoryObj} from '@storybook/react-native';
import {Dimensions} from 'react-native';

import AutoLoopCarouselExample from './Example/AutoLoopCarouselExample';

import {ICarouselAlign} from '@lad-tech/mobydick-core';

const {width: WIDTH, height: HEIGHT} = Dimensions.get('screen');

const meta: Meta<typeof AutoLoopCarouselExample> = {
  title: 'Other',
  component: AutoLoopCarouselExample,
};

type Story = StoryObj<typeof AutoLoopCarouselExample>;

export const AutoLoopCarousel = {
  args: {
    sideMargin: 5,
    activeItemId: 1,
    dataLength: 3,
    isDots: false,
    align: ICarouselAlign.start,
    timerAuto: 2000,
    itemWidth: WIDTH - 10,
    itemHeight: HEIGHT,
    animateAutoScroll: true,
  },
  argTypes: {
    align: {
      options: Object.values(ICarouselAlign),
      control: {
        type: 'select',
      },
    },
    isDots: {
      control: {
        type: 'boolean',
      },
    },
    animateAutoScroll: {
      control: {
        type: 'boolean',
      },
    },
  },
} satisfies Story;

export default meta;
