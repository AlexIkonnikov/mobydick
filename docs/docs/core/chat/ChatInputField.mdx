import ChatInput from './assets/ChatInput.png';

<img src={ChatInput} />

Works with ChatInput component


## Example

```tsx
import {useState} from 'react';

import {
  ChatInputField
} from '@lad-tech/mobydick-core';

export const ChatWidget = () => {
  const [input, setInput] = useState('');

  return (
    <ChatInputField value={input} onChangeText={setInput} />
  );
};

```

## `Props`

### [`TextInput Props`](https://reactnative.dev/docs/textinput#props)

Inherits [TextInput Props](https://reactnative.dev/docs/textinput#props).

