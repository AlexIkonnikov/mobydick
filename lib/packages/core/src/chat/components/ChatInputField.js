"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_native_1 = require("react-native");
const styles_1 = require("../../styles");
const InputField_1 = __importDefault(require("../../inputs/components/InputField/InputField"));
const useStyles_1 = __importDefault(require("../../styles/hooks/useStyles"));
const other_1 = require("../../other");
const px_1 = __importDefault(require("../../styles/utils/px"));
const ChatInputField = (props) => {
    const { textInputContainerStyle, style, containerStyle, ...otherProps } = props;
    const [styles] = (0, useStyles_1.default)(stylesCreate);
    return (<InputField_1.default accessibilityLabel={other_1.LABELS.chatInputField} textInputContainerStyle={[styles.inputContainer, textInputContainerStyle]} multiline={true} style={[styles.textInput, style]} containerStyle={[styles.container, containerStyle]} {...otherProps}/>);
};
const stylesCreate = (0, styles_1.createStyles)(({ spaces }) => ({
    inputContainer: {
        minHeight: (0, px_1.default)(38),
        maxHeight: (0, px_1.default)(196),
        width: '100%',
        borderWidth: undefined,
        marginVertical: 0,
        alignItems: 'center',
        paddingVertical: react_native_1.Platform.select({
            android: spaces.Space4,
            ios: spaces.Space8,
        }),
    },
    container: {
        flex: 1,
        flexDirection: 'column',
        minWidth: undefined,
        marginRight: spaces.Space2,
    },
    textInput: {
        paddingTop: 0,
    },
}));
exports.default = ChatInputField;
