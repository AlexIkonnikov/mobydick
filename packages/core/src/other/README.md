## @npm/mobydick-core/other

## ***Counter***

Счетчик с 1 до 99+

Пропсы:
type — primary, secondary, tertiary, tertiary, destructive;
count — число;
без count — будет нарисован indicator c цветом ElementAttention;
style — можно передать свой backgroundColor, и через top, right, bottom, left — расположить в нужном месте, по дефолту position: absolute;
size — medium(24), small(18), по default — medium;
maxLength —— по default 2, отображает максимальное заданное количество чисел;

## ***CrossedText***

Использовать, когда цвет зачеркнутой линии отличается от цвета текста, в противном случае используйте дефолтный пропс textDecorationLine: 'line-through'.

## ***Collapsible***

Не используйте `typeAnimation`: `spring` и `keyboard`, происходит краш приложения на android;
