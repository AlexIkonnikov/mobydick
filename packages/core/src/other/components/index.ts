export * from './Badge/';

export * from './Dots';
export * from './Avatar';
