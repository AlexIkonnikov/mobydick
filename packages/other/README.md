## @npm/mobydick-other

## ***Counter***

Счетчик с 1 до 99+

Пропсы:
type — primary, secondary, tertiary, tertiary, destructive;
count — число;
style — можно передать свой backgroundColor, и через top, right, bottom, left — расположить в нужном месте, по дефолту position: absolute;
size — medium(24), small(18), по default — medium;

