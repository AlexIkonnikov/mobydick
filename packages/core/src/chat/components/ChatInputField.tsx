import React from 'react';
import {StyleSheet} from 'react-native';

import {ITextInputProps} from '../../basic';
import {rem} from '../../styles';
import InputField from '../../inputs/components/InputField/InputField';
import useStyles from '../../styles/theme/hooks/useStyles';
import {LABELS} from '../../other';

const ChatInputField = (props: ITextInputProps) => {
  const [styles] = useStyles(stylesCreate);

  return (
    <InputField
      accessibilityLabel={LABELS.chatInputField}
      textInputContainerStyle={styles.containerStyle}
      multiline={true}
      style={styles.textInputStyle}
      {...props}
    />
  );
};

const stylesCreate = () =>
  StyleSheet.create({
    containerStyle: {
      minHeight: rem(36),
      maxHeight: rem(196),
      width: rem(267),
      borderWidth: undefined,
    },
    textInputStyle: {
      paddingTop: 0,
    },
  });

export default ChatInputField;
