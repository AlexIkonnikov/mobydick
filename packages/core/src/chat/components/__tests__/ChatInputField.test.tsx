import {act, render} from '@testing-library/react-native';
import React from 'react';

import ChatInputField from '../ChatInputField';
import {LABELS} from '../../../other';

describe('@lad-tech/mobydick-core/ChatInputField', () => {
  it('renders correctly', () => {
    const {toJSON} = render(
      <ChatInputField placeholder={'placeholder'} value={'value'} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly with props', () => {
    const {toJSON} = render(
      <ChatInputField
        placeholder={'placeholder'}
        value={'value'}
        textInputContainerStyle={{width: 200}}
        style={{paddingTop: 20}}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly focused/blur', () => {
    const {toJSON, getByLabelText} = render(<ChatInputField />);
    const TextInput = getByLabelText(LABELS.chatInputField);
    act(() => TextInput.props.onFocus());

    expect(toJSON()).toMatchSnapshot();

    act(() => TextInput.props.onBlur());

    expect(toJSON()).toMatchSnapshot();
  });
});
