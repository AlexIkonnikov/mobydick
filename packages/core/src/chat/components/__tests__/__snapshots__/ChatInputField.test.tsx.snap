// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@lad-tech/mobydick-core/ChatInputField renders correctly 1`] = `
<View
  style={
    [
      {
        "minWidth": 195,
      },
      [
        {
          "flex": 1,
          "flexDirection": "column",
          "marginRight": 3,
          "minWidth": undefined,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "flex-start",
          "backgroundColor": "#EEEFF1",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 12,
          "paddingLeft": 27,
          "paddingRight": 24,
          "paddingVertical": 18,
        },
        {
          "minHeight": 120,
        },
        [
          {
            "alignItems": "center",
            "borderWidth": undefined,
            "marginVertical": 0,
            "maxHeight": 294,
            "minHeight": 57,
            "paddingVertical": 12,
            "width": "100%",
          },
          undefined,
        ],
      ]
    }
  >
    <TextInput
      accessibilityLabel="chatInputField"
      allowFontScaling={true}
      editable={true}
      multiline={true}
      numberOfLines={1}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholder="placeholder"
      placeholderTextColor="#9BA1B0"
      selectionColor="#2B78EE"
      style={
        [
          {
            "flex": 1,
            "padding": 0,
          },
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 24,
            "lineHeight": 30,
            "minHeight": 30,
          },
          [
            {
              "paddingTop": 0,
            },
            undefined,
          ],
        ]
      }
      testID="TextInput"
      value="value"
    />
  </View>
</View>
`;

exports[`@lad-tech/mobydick-core/ChatInputField renders correctly focused/blur 1`] = `
<View
  style={
    [
      {
        "minWidth": 195,
      },
      [
        {
          "flex": 1,
          "flexDirection": "column",
          "marginRight": 3,
          "minWidth": undefined,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "flex-start",
          "backgroundColor": "#EEEFF1",
          "borderColor": "rgba(32, 36, 45, 0.3)",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 12,
          "paddingLeft": 27,
          "paddingRight": 24,
          "paddingVertical": 18,
        },
        {
          "minHeight": 120,
        },
        [
          {
            "alignItems": "center",
            "borderWidth": undefined,
            "marginVertical": 0,
            "maxHeight": 294,
            "minHeight": 57,
            "paddingVertical": 12,
            "width": "100%",
          },
          undefined,
        ],
      ]
    }
  >
    <TextInput
      accessibilityLabel="chatInputField"
      allowFontScaling={true}
      editable={true}
      multiline={true}
      numberOfLines={1}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#9BA1B0"
      selectionColor="#2B78EE"
      style={
        [
          {
            "flex": 1,
            "padding": 0,
          },
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 24,
            "lineHeight": 30,
            "minHeight": 30,
          },
          [
            {
              "paddingTop": 0,
            },
            undefined,
          ],
        ]
      }
      testID="TextInput"
    />
  </View>
</View>
`;

exports[`@lad-tech/mobydick-core/ChatInputField renders correctly focused/blur 2`] = `
<View
  style={
    [
      {
        "minWidth": 195,
      },
      [
        {
          "flex": 1,
          "flexDirection": "column",
          "marginRight": 3,
          "minWidth": undefined,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "flex-start",
          "backgroundColor": "#EEEFF1",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 12,
          "paddingLeft": 27,
          "paddingRight": 24,
          "paddingVertical": 18,
        },
        {
          "minHeight": 120,
        },
        [
          {
            "alignItems": "center",
            "borderWidth": undefined,
            "marginVertical": 0,
            "maxHeight": 294,
            "minHeight": 57,
            "paddingVertical": 12,
            "width": "100%",
          },
          undefined,
        ],
      ]
    }
  >
    <TextInput
      accessibilityLabel="chatInputField"
      allowFontScaling={true}
      editable={true}
      multiline={true}
      numberOfLines={1}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#9BA1B0"
      selectionColor="#2B78EE"
      style={
        [
          {
            "flex": 1,
            "padding": 0,
          },
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 24,
            "lineHeight": 30,
            "minHeight": 30,
          },
          [
            {
              "paddingTop": 0,
            },
            undefined,
          ],
        ]
      }
      testID="TextInput"
    />
  </View>
</View>
`;

exports[`@lad-tech/mobydick-core/ChatInputField renders correctly with props 1`] = `
<View
  style={
    [
      {
        "minWidth": 195,
      },
      [
        {
          "flex": 1,
          "flexDirection": "column",
          "marginRight": 3,
          "minWidth": undefined,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "flex-start",
          "backgroundColor": "#EEEFF1",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 12,
          "paddingLeft": 27,
          "paddingRight": 24,
          "paddingVertical": 18,
        },
        {
          "minHeight": 120,
        },
        [
          {
            "alignItems": "center",
            "borderWidth": undefined,
            "marginVertical": 0,
            "maxHeight": 294,
            "minHeight": 57,
            "paddingVertical": 12,
            "width": "100%",
          },
          {
            "width": 200,
          },
        ],
      ]
    }
  >
    <TextInput
      accessibilityLabel="chatInputField"
      allowFontScaling={true}
      editable={true}
      multiline={true}
      numberOfLines={1}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholder="placeholder"
      placeholderTextColor="#9BA1B0"
      selectionColor="#2B78EE"
      style={
        [
          {
            "flex": 1,
            "padding": 0,
          },
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 24,
            "lineHeight": 30,
            "minHeight": 30,
          },
          [
            {
              "paddingTop": 0,
            },
            {
              "paddingTop": 20,
            },
          ],
        ]
      }
      testID="TextInput"
      value="value"
    />
  </View>
</View>
`;
